packages/devtools_app/lib/src/bloc/bloc/bloc_list_bloc.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../eval_on_dart_library.dart';
+import '../../globals.dart';
+import '../models/bloc_object.dart';
+
+part 'bloc_list_event.dart';
+part 'bloc_list_state.dart';
+
+class BlocListBloc extends Bloc<BlocListEvent, BlocListState> {
+  BlocListBloc(this._evalOnDartLibrary) : super(BlocListLoadInProgress()) {
+    _postEventSubscription =
+        serviceManager.service.onExtensionEvent.where((event) {
+      return event.extensionKind == 'bloc:bloc_map_changed';
+    }).listen((_) {
+      add(BlocListRequested());
+    });
+    add(BlocListRequested());
```

Comment:
In my opinion we should add this event when the bloc is instantiated via `BlocProvider`

```dart
BlocProvider(
  create: (_) => BlocListBloc()..add(BlocListRequested()),
  ...
),
```

This keeps the bloc flexible, makes it easier to test, and is generally more explicit.